### ISSUE TITLE
Web editor indentation

### ISSUE THREAD

**Issue by joubertredrat** [gogits/gogs#3610]
Now with web editor working fine, I think that is necessary to set configurations by default or defined by user for better experience with web editor.

On PHP, javascript and I think that Golang (@Unknwon talk me if I'm wrong) we prefer 4 spaces indentation.
However, HTML and CSS developers normally prefer 2 spaces indentation. Have cases on other developers prefer tabulation instead spaces.

Because this I think that web editor can be configurable, one good example for this is on Bitbucket, you can set options in own editor.

![screenshot-bitbucket org 2016-09-06 13-25-57](https://cloud.githubusercontent.com/assets/1520407/18281998/8599e9a6-7435-11e6-94a4-bc4492e2ee33.png)

Or this configuration can be defined on user settings on only on admin area.

What you think guys?

Reference: https://github.com/gogits/gogs/issues/3460#issuecomment-242807939

**Comment by lavvy** [comment]
yes, please we really need to consider this issue more seriously , Many times When you edit Some Makefiles on gogs web editor it will keep throwing errors when building . But when do the same change on the the same code in github or gitlab , it builds without error. Its really serious.

**Comment by DennyLoko** [comment]
I think it should follow the [`.editorconfig`](http://editorconfig.org/) spec.

**Comment by rugk** [comment]
AFAIK editorconfig support is already implemented. However I would still be in favour of this issue as it is just a "usual" setting, which you can e.g. also find on GItHub.

**Comment by Unknwon** [comment]
@lavvy thanks your comment but you basically said nothing..

**Comment by lavvy** [comment]
@unknown i had experienced situations when the repo i edited in gogs throws up errors. I found out that it seems like gogs kind of rearrange the codes internally. For instance some makefiles are very particular about tabs,  but when you put spaces it throws up errors. I  can't remember the particular case i had then. But when migrated the same code to github or gitlab it built. I feel more comfortable working on gogs but that issue was always frustrating. I think it has to do with spaces and tabs.

**Comment by lavvy** [comment]
@unknown i have been able to catch that bug again. It has to do with CRLF line ending. You can reproduce it too from my issue , see the issue for it here at #3898
